common/arch/sdl/joy.cpp

```diff
@@ -372,10 +376,19 @@ void joy_init()
 			const auto n_buttons = check_warn_joy_support_limit(SDL_JoystickNumButtons(handle), "button", DXX_MAX_BUTTONS_PER_JOYSTICK);
 			const auto n_hats = check_warn_joy_support_limit(SDL_JoystickNumHats(handle), "hat", DXX_MAX_HATS_PER_JOYSTICK);
 
-			joybutton_text.resize(joybutton_text.size() + n_buttons + (4 * n_hats) + (2 * n_axes));
+			const auto n_virtual_buttons = n_buttons + (4 * n_hats) + (2 * n_axes);
+			joybutton_text.resize(joybutton_text.size() + n_virtual_buttons);
+			joy_key_map.resize(joy_key_map.size() + n_virtual_buttons, 0);
```

Comment:
The single argument form of `std::vector<T>::resize` would zero-initialize the new members, so there is no need to explicitly use 0.

No need to change this.

Reply:
I thought the same, but in the end opted for "better explicit than imprecise".